airflow/providers/exasol/hooks/exasol.py

```diff
@@ -159,19 +159,21 @@ def run(
         """
         if isinstance(sql, str):
             if split_statements:
-                sql = self.split_sql_string(sql)
+                sql_list: Iterable[str] = self.split_sql_string(sql)
```

Comment:
The root cause of the problem is line https://github.com/apache/airflow/pull/27997/files#diff-00b215634013629b320a263630bf6948ed33c2a8528754fc78d69484e1320845R191 

the`sql` parameter changed it type - when it was passed as "string", it was always converted to list and that caused the return to be wrapped in list as well. So when `sql` was used in `return_single_query_results`, it was already a list and return_single_query_resuts is always false when you pass list of queries.